plugins/inputs/ravendb/ravendb.go

```diff
@@ -0,0 +1,434 @@
+
+	err := r.requestJSON(r.requestUrlIndexes, &indexesResponse)
+	if err != nil {
+		acc.AddError(err)
+		return
+	}
+
+	for _, perDbIndexResponse := range indexesResponse.Results {
+		for _, indexResponse := range perDbIndexResponse.Indexes {
+			tags := map[string]string{
+				"database_name": perDbIndexResponse.DatabaseName,
+				"index_name":    indexResponse.IndexName,
+				"node_tag":      indexesResponse.NodeTag,
+				"url":           r.URL,
+			}
+
+			if indexesResponse.PublicServerUrl != nil {
+				tags["public_server_url"] = *indexesResponse.PublicServerUrl
+			}
+
+			fields := map[string]interface{}{
+				"errors":                          indexResponse.Errors,
+				"is_invalid":                      indexResponse.IsInvalid,
+				"lock_mode":                       indexResponse.LockMode,
+				"mapped_per_sec":                  indexResponse.MappedPerSec,
+				"priority":                        indexResponse.Priority,
+				"reduced_per_sec":                 indexResponse.ReducedPerSec,
+				"state":                           indexResponse.State,
+				"status":                          indexResponse.Status,
+				"time_since_last_indexing_in_sec": indexResponse.TimeSinceLastIndexingInSec,
+				"time_since_last_query_in_sec":    indexResponse.TimeSinceLastQueryInSec,
+				"type":                            indexResponse.Type,
+			}
+
+			acc.AddFields("ravendb_indexes", fields, tags)
+		}
+	}
+}
+
+func (r *RavenDB) gatherCollections(acc telegraf.Accumulator) {
+	collectionsResponse := &collectionsMetricResponse{}
+
+	err := r.requestJSON(r.requestUrlCollection, &collectionsResponse)
+	if err != nil {
+		acc.AddError(err)
+		return
+	}
+
+	for _, perDbCollectionMetrics := range collectionsResponse.Results {
+		for _, collectionMetrics := range perDbCollectionMetrics.Collections {
+			tags := map[string]string{
+				"collection_name": collectionMetrics.CollectionName,
+				"database_name":   perDbCollectionMetrics.DatabaseName,
+				"node_tag":        collectionsResponse.NodeTag,
+				"url":             r.URL,
+			}
+
+			if collectionsResponse.PublicServerUrl != nil {
+				tags["public_server_url"] = *collectionsResponse.PublicServerUrl
+			}
+
+			fields := map[string]interface{}{
+				"documents_count":          collectionMetrics.DocumentsCount,
+				"documents_size_in_bytes":  collectionMetrics.DocumentsSizeInBytes,
+				"revisions_size_in_bytes":  collectionMetrics.RevisionsSizeInBytes,
+				"tombstones_size_in_bytes": collectionMetrics.TombstonesSizeInBytes,
+				"total_size_in_bytes":      collectionMetrics.TotalSizeInBytes,
+			}
+
+			acc.AddFields("ravendb_collections", fields, tags)
+		}
+	}
+}
+
+func prepareDbNamesUrlPart(dbNames []string) string {
+	if len(dbNames) == 0 {
+		return ""
+	}
+	result := "?" + dbNames[0]
+	for _, db := range dbNames[1:] {
+		result += "&name=" + url.QueryEscape(db)
+	}
+
+	return result
+}
+
+func (r *RavenDB) Init() error {
+	if r.URL == "" {
+		r.URL = defaultURL
+	}
```

Comment:
I believe instead of this block, you can just set the URL to defaultURL in the init() method, thus default will be used unless the user configures it.

Reply:
No please leave this in here to prevent user explicitly specifying `url = ""` from being surprised.

Reply:
leaving as it is.